explorer/testdata/class/fail_use_before_typecheck.carbon

```diff
@@ -0,0 +1,19 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{not} %{explorer} %s 2>&1 | %{FileCheck-strict} %s
+// RUN: %{not} %{explorer-trace} %s 2>&1 | %{FileCheck-allow-unmatched} %s
+// AUTOUPDATE: %{explorer} %s
+
+package ExplorerTest api;
+
+class C {
+  // CHECK:COMPILATION ERROR: {{.*}}/explorer/testdata/class/fail_use_before_typecheck.carbon:[[@LINE+1]]: incomplete type `class C` used in member access
```

Comment:
Is the test I suggested on #1394 distinct?

```
class Foo {
  fn PickType() -> Type { return i32; }
  fn GetZero() -> PickType() { return 0; }
}
```

Reply:
Yeah, that seems worth testing separately, especially given the "delay methods until after the class" rule. Added.